File: Services/java-services/Core/CoreShared/src/main/java/ai/omnicure/core/shared/pagination/PagedList.java
package ai.omnicure.core.shared.pagination;

import lombok.Getter;

import java.util.List;

@Getter
public class PagedList<T> {

    private final int currentPage;
    private final int totalPages;
    private final int pageSize;
    private final long totalCount;
    private final boolean hasPrevious;
    private final boolean hasNext;
    private final List<T> items;

    public PagedList(List<T> items, long count, int pageNumber, int pageSize) {
        this.items = items;
        this.totalCount = count;
        this.currentPage = pageNumber;
        this.pageSize = pageSize;
        this.totalPages = (int) Math.ceil((double) count / pageSize);
        this.hasPrevious = currentPage > 1;
        this.hasNext = currentPage < totalPages;
    }

    public static <T> PagedList<T> of(List<T> items, long count, int pageNumber, int pageSize) {
        return new PagedList<>(items, count, pageNumber, pageSize);
    }
}

File: Services/dotnet-services/Core/OmniCure.Core.Shared/Pagination/PagedList.cs
namespace OmniCure.Core.Shared.Pagination;

public class PagedList<T>(List<T> items, int count, int pageNumber, int pageSize)
{
    public int CurrentPage { get; private set; } = pageNumber;

    public int TotalPages { get; private set; } = (int)Math.Ceiling(count / (double)pageSize);

    public int PageSize { get; private set; } = pageSize;

    public int TotalCount { get; private set; } = count;

    public bool HasPrevious => CurrentPage > 1;

    public bool HasNext => CurrentPage < TotalPages;

    public List<T> Items { get; private set; } = items;
}
